llvm-bazel/llvm-project-overlay/llvm/include/llvm/Config/llvm-config.h

```diff
@@ -68,7 +68,7 @@
 #define LLVM_USE_OPROFILE 0
 
 /* Define if we have the perf JIT-support library */
-#define LLVM_USE_PERF 0
+#define LLVM_USE_PERF 1
```

Comment:
I've hit distros where this isn't available... is it important to change this default?

We could move this to the configured macro set like we do for some other libraries where we support both using and not using them.

Reply:
Yeah I figured this would be the most controversial thing. We can move it to something more configurable. One of the wrinkles is that we have to actually not add PerfJITEventListener.cpp to the srcs if this set off, which is a bit annoying, because of a redefinition of createPerfJITEventListener:

> ERROR: /usr/local/google/home/gcmn/.cache/bazel/_bazel_gcmn/1655d0ec87ba8569e84422cbb0aeec46/external/llvm-project/llvm/BUILD:1996:11: C++ compilation of rule
 '@llvm-project//llvm:ExecutionEngine' failed (Exit 1)                                                                                                        
external/llvm-project/llvm/lib/ExecutionEngine/PerfJITEvents/PerfJITEventListener.cpp:495:37: error: redefinition of 'createPerfJITEventListener'             
JITEventListener *JITEventListener::createPerfJITEventListener() {                                                                                            
                                    ^                                                                                                                         
external/llvm-project/llvm/include/llvm/ExecutionEngine/JITEventListener.h:102:28: note: previous definition is here                                          
  static JITEventListener *createPerfJITEventListener()  

In CMake, this is gated with an add_subdirectory. We should probably break up the ExecutionEngine target, but I have this sneaking suspicion it is the way it is because of bad layering. Thoughts?

Reply:
Also looks like this is strictly only supported on linux, so we should at least gate it on that

Reply:
Ooof.

I don't see a good way of doing this other than breaking up `ExecutionEngine` so that we can avoid building the code when this is missing.

I'd suggest just keeping this internal until that refactoring can land.

I'm hopeful the layering isn't too bad though -- my guess is we just hadn't gotten to the point of breaking it up.

Reply:
Ok I'll do that.